engine/flink/management/src/main/scala/pl/touk/nussknacker/engine/management/FlinkRestManager.scala

```diff
@@ -58,17 +60,48 @@ class FlinkRestManager(config: FlinkConfig, modelData: ModelData, sender: HttpSe
 
   private val flinkUrl = dispatch.url(config.restUrl)
 
-  private lazy val uploadedJarId : Future[String] = uploadCurrentJar()
+  // after job manager restart old resources are not available anymore and we have to upload jar once again
+  private var jarUploadedBeforeLastRestart: Option[Future[String]] = None
```

Comment:
I wonder if some synchronization would be needed here? Or we rely on ManagementActor which prevents concurrent deploys? Then maybe a comment why we don't do it here?

Reply:
Yeah, everything is synchronous here (at least for now). But I see your point, I've introduced a synchronized block - we'd better be safe than sorry.

Reply:
I think for now better add just comment - synchronized is no longer enough, since we do the check asynchronously?

Reply:
Right, totally missed the fact that updates are now done within a Future.